server/files/api_views.py

```diff
@@ -21,14 +22,24 @@ def perform_destroy(self, instance):
 
     def create(self, request):
         (metadata, file) = (json.loads(self.request.data["metadata"]), self.request.data["file"])
+
+        notebook = Notebook.objects.get(id=metadata["notebook_id"])
```

Comment:
The `get()` queryset method raises `DoesNotExist` or `MultipleObjectsReturned` in case there are no items or more than one (the latter being harder here since it queries for a primary key here). I would suggest to use `get_object_or_404` here to explicitly handling `DoesNotExist` here.